ci-operator/step-registry/cucushift/chainupgrade/toimage/cucushift-chainupgrade-toimage-commands.sh

```diff
@@ -542,6 +542,152 @@ function check_history() {
+    extrainfoCmd="oc get infrastructure cluster -o yaml | yq '.status'"
+    if [[ -n "$platform" ]] ; then
+        case "$platform" in
+            none|powervs)
+                export E2E_RUN_TAGS="@baremetal-upi and ${E2E_RUN_TAGS}"
+                eval "$extrainfoCmd"
+                ;;
+            external)
+                echo "Expected, got platform as '$platform'"
+                eval "$extrainfoCmd"
+                ;;
+            alibabacloud)
+                export E2E_RUN_TAGS="@alicloud-${ipixupi} and ${E2E_RUN_TAGS}"
+                ;;
+            aws|azure|baremetal|gcp|ibmcloud|nutanix|openstack|vsphere)
+                export E2E_RUN_TAGS="@${platform}-${ipixupi} and ${E2E_RUN_TAGS}"
+                ;;
+            *)
+                echo "Unexpected, got platform as '$platform'"
+                eval "$extrainfoCmd"
+                ;;
+        esac
+    fi
+    echo_e2e_tags
+}
+
+function filter_test_by_proxy() {
+    local proxy
+    proxy="$(oc get proxies.config.openshift.io cluster -o yaml | yq '.spec|(.httpProxy,.httpsProxy)' | uniq)"
+    if [[ -n "$proxy" ]] && [[ "$proxy" != 'null' ]] ; then
+        export E2E_RUN_TAGS="@proxy and ${E2E_RUN_TAGS}"
+    fi
+    echo_e2e_tags 
+}
+
+function filter_test_by_fips() {
+    local data
+    data="$(oc get configmap cluster-config-v1 -n kube-system -o yaml | yq '.data')"
+    if ! (grep --ignore-case --quiet 'fips' <<< "$data") ; then
+        export E2E_RUN_TAGS="not @fips and ${E2E_RUN_TAGS}"
+    fi
+    echo_e2e_tags
+}
+
+function filter_test_by_sno() {
+    local nodeno
+    nodeno="$(oc get nodes --no-headers | wc -l)"
+    if [[ $nodeno -eq 1 ]] ; then
+        export E2E_RUN_TAGS="@singlenode and ${E2E_RUN_TAGS}"
+    fi
+    echo_e2e_tags 
+}
+
+function filter_test_by_network() {
+    local networktype
+    networktype="$(oc get network.config/cluster -o yaml | yq '.spec.networkType')"
+    case "${networktype,,}" in
+        openshiftsdn)
+        networktag='@network-openshiftsdn'
+        ;;
+        ovnkubernetes)
+        networktag='@network-ovnkubernetes'
+        ;;
+        *)
+        echo "######Expected network to be SDN/OVN, but got: $networktype"
+        ;; 
+    esac
+    if [[ -n $networktag ]] ; then
+        export E2E_RUN_TAGS="${networktag} and ${E2E_RUN_TAGS}"
+    fi
+    echo_e2e_tags
+}
+
+
+
+function filter_tests() {
+    filter_test_by_fips
+    filter_test_by_proxy
+    filter_test_by_sno
+    filter_test_by_network
+    filter_test_by_platform
+    filter_test_by_version
+
+    echo_e2e_tags
+}
+
+function run_upgrade_e2e() {
+    local PARALLEL=4
+    local E2E_SKIP_TAGS="not @console
+          and not @destructive
+          and not @disconnected
+          and not @flaky
+          and not @inactive
+          and not @prod-only
+          and not @stage-only
+          and not @upgrade-check
+          and not @upgrade-prepare
+          and not @serial
+    "
+
+    E2E_RUN_TAGS="$E2E_RUN_TAGS and $E2E_SKIP_TAGS"
```

Comment:
How about use `E2E_SANITY_TAGS`, so that it distinguish with the "real" `E2E_RUN_TAGS` ?

Reply:
I was thinking keep the naming of `E2E_RUN_TAGS`, because what suite of e2e is executed is decided by its value (current it is `@upgrade-sanity`), if in the future, we need more test suite, we just need to update its default value, but not have to update the var name together.